compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -407,6 +407,12 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
     def newTupleMirror(arity: Int): Tree =
       New(defn.RuntimeTupleMirrorTypeRef, Literal(Constant(arity)) :: Nil)
 
+    def newJavaRecordReflectMirror(tpe: Type) =
```

Comment:
This mirror is of limited use and requires the user to check the type at the derivation site for the mirrored mono type. This isn't the end of the world - we could add an extension method which uses the `RecordComponent` to invoke the accessor, but can we add a method to the mirror like `productElement(a: A)(i: Int)` which allows users which derive with the mirror to hide the implementation details?